thecourseforum/theCourseForum2
# Extra Info Form Validation/UI

## Status
- Done

## GitHub Issues addressed (`#IssueNumber`)
- This PR closes #208 

## What I did
- Added graduation year validation to extra info form (when creating new account)
- Default graduation year set to the current year
- Fixed form styling to use bootstrap

## Screenshots
- Before
<img width="516" alt="before" src="https://user-images.githubusercontent.com/34071587/108609889-15f05b80-739f-11eb-8680-62d3733a4cc4.PNG">

- After
<img width="537" alt="after2" src="https://user-images.githubusercontent.com/34071587/109091606-1eb89880-76e3-11eb-877f-b0d401dbcd30.PNG">

(screenshots below are outdated but the validation pretty much works the same)
<img width="559" alt="small" src="https://user-images.githubusercontent.com/34071587/108609896-21438700-739f-11eb-8e88-3176e175f1fe.PNG">
<img width="514" alt="large" src="https://user-images.githubusercontent.com/34071587/108609897-243e7780-739f-11eb-9684-eb0e1302b24c.PNG">

## Tests
- A brief explanation of tests done/written or how reviewers can test your work
1. Delete account in Django admin*
2. Sign in
3. Input graduation year too small (negative, 0, <1900) and/or too large (<current year + 6), click submit
4. Input invalid number (decimal, nonnumerical characters), click submit
5. Input graduation year in appropriate range, click submit
6. Check profile page to verify correct graduation year is entered

*alternatively go to endpoint `/login/collect_extra_info` to test the form without deleting your account (the site will crash if the form is submitted though)

## Questions/Discussions/Notes
- as mentioned above it's possible to access the `/login/collect_extra_info` endpoint and crash the site if you already have an account, not sure this is really a big issue though

## Merging the PR
- Who should merge this PR?
- Should the commit history be preserved in the base branch, or is it okay to combine all of them into a single commit ("squash-merge")?
